aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cpp

```diff
@@ -350,136 +351,126 @@ Tensor& _linalg_inv_out_helper_cuda_lib(Tensor& result, Tensor& infos_getrf, Ten
-  auto self_stride = matrixStride(self);
-  auto U_stride = !compute_uv ? 1 : matrixStride(U);
+  auto A_stride = matrixStride(A);
   auto S_stride = S.size(-1);
 
-  int _batchsize = cuda_int_cast(batchCount(self), "batch size");
-  int m = cuda_int_cast(self.size(-2), "m");
-  int n = cuda_int_cast(self.size(-1), "n");
+  int m = cuda_int_cast(A.size(-2), "m");
+  int n = cuda_int_cast(A.size(-1), "n");
+  auto k = std::min(m, n);
   int lda = std::max<int>(1, m);
-  int ldvt = std::max<int>(1, n);
+  int ldvh = std::max<int>(1, n);
 
   TORCH_INTERNAL_ASSERT(m >= n, "cusolver gesvd only supports matrix with sizes m >= n");
 
-  char job = compute_uv ? (some ? 'S' : 'A') : 'N';
+  char job = compute_uv ? (full_matrices ? 'A' : 'S') : 'N';
   auto handle = at::cuda::getCurrentCUDASolverDnHandle();
 
   int lwork = -1;
   at::cuda::solver::gesvd_buffersize<scalar_t>(handle, m, n, &lwork);
   TORCH_INTERNAL_ASSERT(lwork >= 0, "gesvd_buffersize failed to get needed buffer size, got lwork = ", lwork);
 
   auto& allocator = *::c10::cuda::CUDACachingAllocator::get();
-  auto dataPtr_work = allocator.allocate(sizeof(scalar_t)*lwork);
-  auto dataPtr_rwork = allocator.allocate(sizeof(value_t)*std::min(m, n));
+  const auto dataPtr_work = allocator.allocate(sizeof(scalar_t)*lwork);
+  const auto dataPtr_rwork = allocator.allocate(sizeof(value_t)*std::min(m, n));
 
-  // VT_view shares the same memory space as VT tensor
-  at::Tensor VT_view;
-  if (compute_uv) {
-    VT_view = VT.view({-1, n, n});
-  }
+  // nb. We can do this .view() because V is a batch of F-contig matrices
+  const auto V_view = compute_uv ? V.view({-1, n, V.size(-1)})
+                                 : Tensor{};
+  // V is F-contig. Since this function computes Vh, we need an auxiliary F-conj-transposed matrix to hold Vh
+  const auto Vh_workspace = compute_uv ?  at::empty({n, full_matrices ? n : k},
+                                              A.options().memory_format(at::MemoryFormat::Contiguous)).conj()
+                                       : Tensor{};
+  const auto Vh_ptr = compute_uv ? Vh_workspace.data_ptr<scalar_t>()
+                                 : nullptr;
+
+  const auto U_stride = compute_uv ? matrixStride(U) : 0;
+  const auto U_ptr = compute_uv ? U.data_ptr<scalar_t>() : nullptr;
+
+  int batchsize = calculate_all_batches ? cuda_int_cast(batchCount(A), "batch size")
+                                        : batches.size();
 
-  // This function `gesvd` mainly serves as a fallback of `gesvdj` when the latter failed to converge for some large matrices.
-  // `gesvdj` returns matrix V in original form of V, while `gesvd` returns matrix V as Hermitian conjugate V^H.
-  // With that, we can't directly write `gesvd` output into the memory of tensor VT, which has preset strides and conjugate flag
-  // of V, not V^H.
-  // One solution is to use an intermediate buffer vt_workspace to get V^H output from `gesvd`, then copy it to the result
```

Comment:
I figured that the more concise 
>  // V is F-contig. Since this function computes Vh, we need an auxiliary F-conj-transposed matrix to hold Vh 

could do?

Reply:
I don't know whether we need conciseness in the comments.
This function differs from others in that it computes the result only for selected matrices in the batch and uses a single workspace memory that is reused in the for loop, so I feel that all this information written here is indeed useful to understand what's going on.

Hopefully, after your rewrite nobody ever would need to look at the code, so it's fine to remove the comment.